File: app/src/main/java/com/earth/OsToolkit/SplashActivity.java
package com.earth.OsToolkit;

import android.Manifest;
import android.app.Activity;
import android.content.*;
import android.os.Bundle;
import android.support.v4.app.ActivityCompat;

import static com.earth.OsToolkit.Base.Checking.checkRoot;

@SuppressWarnings("all")
public class SplashActivity extends Activity {

	/*
	 * 27 Dec 2018
	 *
	 * By 1552980358
	 *
	 */

	@Override
	public void onCreate(Bundle savedInstanceSatate) {
		super.onCreate(savedInstanceSatate);
		setContentView(R.layout.activity_welcome);

		SharedPreferences sp = getSharedPreferences("save", MODE_PRIVATE);
		int welcome = sp.getInt("welcome", 0);

		ActivityCompat.requestPermissions(this, new String[]{Manifest.permission.INTERNET,
				Manifest.permission.ACCESS_NETWORK_STATE}, 1);

		Intent intent = new Intent();
		if (welcome == 0 || !checkRoot()) {
			intent.setClass(this, WelcomeActivity.class);
		} else {
			intent.setClass(this, MainActivity.class);
		}

		startActivity(intent);
		this.finish();
	}

}

File: app/src/main/java/aosp/toolkit/osmium/SplashActivity.kt
package aosp.toolkit.osmium

/*
 * OsToolkit - Kotlin
 *
 * Date : 31/12/2018
 *
 * By   : 1552980358
 *
 */

/*
 * Modify
 *
 * 9/1/2019
 *
 */

import android.content.*
import android.os.Bundle
import android.support.v4.app.ActivityCompat
import android.support.v7.app.AppCompatActivity

class SplashActivity : AppCompatActivity() {
    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)

        val permission = arrayOf(android.Manifest.permission.INTERNET)

        ActivityCompat.requestPermissions(this, permission, 1)

        if (getSharedPreferences("splash", Context.MODE_PRIVATE)
                .getBoolean("welcome", false)
        ) {
            startActivity(Intent(this, MainActivity::class.java))
        } else {
            startActivity(Intent(this, WelcomeActivity::class.java))
        }

        finish()
    }
}